templates/src/app/file/index.js

```diff
@@ -1,43 +1,37 @@
 var { Router } = require("express");
 var aws = require("aws-sdk");
-var cfsign = require("aws-cloudfront-sign");
 
-var router = Router()
+var router = Router();
 var s3 = new aws.S3();
 
-router.get("/presigned/:key", (req, res) => {
+router.get("/presigned", (req, res) => {
+  var bucketName = (req.query.bucket)?(req.query.bucket):process.env.AWS_S3_DEFAULT_BUCKET;
   var params = {
-    Bucket: process.env.S3_BUCKET,
-    Fields: {
-      key: req.params.key,
-    },
+    Bucket: bucketName,
+    Key: req.query.key,
+    Expires: 60 * 60, // 60 minutes
   };
-
-  s3.createPresignedPost(params, (err, data) => {
-    if (err) {
-      console.error(err);
-      res.sendStatus(500);
-    } else {
-      console.log(data);
-      res.send(data);
-    }
+  const url = s3.getSignedUrl('putObject',params);
+  console.log(url);
+  res.send({
+    "url":url,
+    "method":"put"
   });
 });
 
-router.get("/:key", (req, res) => {
+router.get("/",(req, res) => {
+  var bucketName = (req.query.bucket)?(req.query.bucket):process.env.AWS_S3_DEFAULT_BUCKET;
   var params = {
-    keypairId: process.env.CF_KEYPAIR_ID,
-    privateKeyString: process.env.CF_KEYPAIR_SECRET_KEY,
-    expireTime: new Date().getTime() + 30000, // defaults to 30s
+    Bucket: bucketName,
+    Key: req.query.key,
+    Expires: 60 * 60, // 60 minutes
   };
-
-  var url = cfsign.getSignedUrl(
-    `https://files.${process.env.DOMAIN}/${req.params.key}`,
-    params
-  );
-
+  const url = s3.getSignedUrl('getObject',params);
```

Comment:
Why was this changed to s3 urls instead of cloudfront?

Reply:
The cloudfront in /file in the previous version didn't return presigned url yet. Using s3 is able to get presigned urls. It looks Cloudfront needs more configuration on AWS. So I temporarily use S3.